x-pack/platform/plugins/shared/index_management/common/lib/template_serialization.ts

```diff
@@ -23,7 +23,10 @@ import {
 
 const hasEntries = (data: object = {}) => Object.entries(data).length > 0;
 
-export function serializeTemplate(templateDeserialized: TemplateDeserialized): TemplateSerialized {
+export function serializeTemplate(
+  templateDeserialized: TemplateDeserialized,
+  dataStreamOptions?: object | undefined
```

Comment:
nit: would it make sense to make this type slightly more specific than just object?

```typescript
export interface DataStreamOptions {
  failure_store?: {
    enabled: boolean;
  };
  [key: string]: unknown;
}
```